src/mode/text_highlight_rules.js

```diff
@@ -99,12 +99,16 @@ TextHighlightRules = function() {
     };
 
     var pushState = function(currentState, stack) {
-        if (currentState != "start" || stack.length)
-            stack.unshift(this.nextState, currentState);
+        if (typeof currentState.get == "function") {
```

Comment:
Can we keep only the scope  variant here?

Reply:
We would need to make significant changes in `normalizeRules` logic in that case